### PR TITLE
Add IPv6 :type and :scope

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/models/inet.py [hunk 1]
```diff
@@ -579,15 +606,19 @@ def _normPyStr(self, valu):
             v6 = ipaddress.IPv6Address(valu)
             v4 = v6.ipv4_mapped
 
-            subs = {'type': getAddrType(v6)}
+            addrtype = getAddrType(v6)
```
Comment (MichaelSquires) [inline synapse/models/inet.py:609]: Looks like this line isn't needed anymore.

File: synapse/models/inet.py [hunk 2]
```diff
@@ -2057,6 +2088,13 @@ def getModelDefs(self):
 
                         ('loc', ('loc', {}), {
                             'doc': 'The geo-political location string for the IPv6.'}),
+
+                        ('type', ('str', {}), {
+                            'doc': 'The type of IP address (e.g., private, multicast, etc.).'}),
+
+                        ('scope', ('str', {}), {
```
Comment (MichaelSquires) [inline synapse/models/inet.py:2095]: Should this be an enum?

**Review by MichaelSquires — CHANGES_REQUESTED** [review]

**Review by MichaelSquires — DISMISSED** [review]

**Review by MichaelSquires — APPROVED** [review]

File: synapse/tests/test_lib_modelrev.py [hunk 3]
```diff
@@ -428,3 +428,9 @@ async def test_modelrev_0_2_22(self):
 
             for node in nodes:
                 self.eq(node.props.get('type'), 'shared')
+
+    async def test_modelrev_0_2_23(self):
+
+        with self.getLoggerStream('synapse.lib.modelrev') as stream:
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_modelrev.py:434]: what is stream doing here?

**Review by vEpiphyte — APPROVED** [review]
We'll need to cordinate the merge of this with the CPE 2.2/2.3 change.

File: synapse/tests/test_lib_modelrev.py (hunk 3 above)
Comment (invisig0th) [reply synapse/tests/test_lib_modelrev.py:434]: This may be vestigial from @MichaelSquires 's edits. I'll remove.

Comment (invisig0th) [reply synapse/tests/test_lib_modelrev.py:434]: ( looks like he already removed it when he separated the modelrev PRs )

**Review by MichaelSquires — APPROVED** [review]

**Pull request merged**